hibernate-core/src/test/java/org/hibernate/orm/test/batch/EagerManyToOneBatchTest.java

```diff
@@ -0,0 +1,123 @@
+package org.hibernate.orm.test.batch;
+
+import java.util.List;
+
+import org.hibernate.cfg.AvailableSettings;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.Setting;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@DomainModel(
+		annotatedClasses = {
+				EagerManyToOneBatchTest.Child.class,
+				EagerManyToOneBatchTest.Parent.class
+		}
+)
+@SessionFactory
+@ServiceRegistry(
+		settings = @Setting(name = AvailableSettings.DEFAULT_BATCH_FETCH_SIZE, value = "2")
+)
+@TestForIssue( jiraKey = "HHH-15346")
+public class EagerManyToOneBatchTest {
+
+	@BeforeEach
+	public void setUp(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Parent parent = new Parent( 1L, "a" );
+					session.persist( parent );
+
+					Child child1 = new Child( 4L, parent );
+					session.persist( child1 );
+
+					Child child2 = new Child( 5L, parent );
+					session.persist( child2 );
+				}
+		);
+	}
+
+	@Test
+	public void testSelect(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					List<Child> children = session
+							.createQuery( "select c from Child as c where c.parent.name=:name", Child.class )
+							.setParameter( "name", "a" )
+							.getResultList();
+
+					assertEquals( 2, children.size() );
+
+					Child child1 = children.get( 0 );
+					assertNotNull( child1.getParent() );
+
+					Child child2 = children.get( 1 );
+					assertNotNull( child2.getParent() );
```

Comment:
Shouldn't we maybe also assert the number of SQL queries that are executed?

Reply:
I added the assertions and also an additional test class for a Lazy association